n_queen_problem/NQueenProblem.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Following code is the implementation of the N-queen problem.
+ * The solution is to place the queens on board such that no queen is under attack by another queen.
+ * There can be a number of possible solutions for a given length of board.
+ * This implementation prints only one valid solution, it can be extended to print all possible valid solutions.
+ */
+
+import java.util.Arrays;
+
+public class NQueenProblem {
+
+    private static boolean isQueenSafe(int[][] board, int row, int column) {
+        int i;
+        int j;
+        int boardLength = board[0].length;
+        for (int i = 0; i < column; ++i)
+            if (board[row][i] == 1)
+                return false;  // If there's another Queen present in the same row.
+
+        for (i = row, j = column; i >= 0 && j >= 0; --i, --j)
+            if (board[i][j] == 1)
```

Comment:
You should make it
 `for (int i = row, j = column; i >= 0 && j >= 0; --i, --j)`
this will reduce the scope of `i` and `j` and fix [this](http://ideone.com/DyurMb) error.